common/src/main/java/tachyon/underfs/UnderFileSystemRegistry.java

```diff
@@ -71,6 +73,7 @@
  * will have the first opportunity to indicate whether they support a requested path.
  * </p>
  */
+@ThreadSafe
```

Comment:
this class is not thread safe, 
see potential race between `register` and `unregister`